File: frontend/src/types/job.ts
export interface AgentProgressStatus {
  name: string;
  status: 'pending' | 'running' | 'completed' | 'failed';
  error?: string | null;
}

export interface JobStatusResponse {
  job_id: string;
  status: 'queued' | 'running' | 'completed' | 'failed' | 'cancelled';
  current_stage: string;
  agents: AgentProgressStatus[];
  error_message?: string | null;
  metadata?: Record<string, any>;
}

export interface SSEJobEvent {
  event: string;
  data: Record<string, any>;
}

File: backend/app/schemas/job.py
from typing import Any, Dict, List, Optional
from pydantic import BaseModel

class AgentProgressStatus(BaseModel):
    name: str
    status: str  # pending, running, completed, failed
    error: Optional[str] = None

class JobStatusResponse(BaseModel):
    job_id: str
    status: str  # queued, running, completed, failed, cancelled
    current_stage: str
    agents: List[AgentProgressStatus] = []
    error_message: Optional[str] = None
    metadata: Dict[str, Any] = {}
